### PR TITLE
[JW8-11793] Fix viewability pause not working during ad buffering

### PR DESCRIPTION
**Pull request by pajong**
### This PR will...
- When the ad state is buffering, wait until the ad state changes to pause from viewability

### Why is this Pull Request needed?
- State change to `pause` is blocked during buffering.
- Instead of unblocking the state change, we want to wait until the playback starts to change to `pause`

#### Addresses Issue(s):

JW8-11793

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

**Comment by jwrobot-player** [comment]
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1839/) MULTI Build for commit 41d17fe4ac35493563015bde29cfbfea774fb195 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9245/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9245/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6206/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6206/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28173/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28173/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23656/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23656/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/5850/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28173/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28173/archive/test/view/) Views

**Review by robwalch — COMMENTED** [review]
~This looks risky~

**Review by robwalch — APPROVED** [review]

File: src/js/controller/controller.js [hunk 1]
```diff
@@ -431,16 +431,34 @@ Object.assign(Controller.prototype, {
         }
 
         function _pauseWhenNotViewable(viewable) {
+            _removeInstreamStateListener();
             if (!viewable) {
-                _this.pause({ reason: 'viewable' });
-                _model.set('playOnViewable', !viewable);
+                if (_getAdState() === STATE_BUFFERING) {
+                    _this._instreamAdapter.once('state', _pauseFromViewable, _this);
```
Comment (robwalch) [inline src/js/controller/controller.js:437]: I understand the need to defer the call to `_this.pause` while buffering, but don't we always want the model "playOnViewable" attribute to be set here?

Other areas in the code code depend on "playOnViewable" being set here regardless of player state.

Comment (robwalch) [reply src/js/controller/controller.js:437]: It seems like this logic could be better handled in `_pause` where we are already checking the ad state and perform custom reasoning around what to do when ad playback is active and not paused.

**Review by robwalch — CHANGES_REQUESTED** [review]

File: src/js/controller/controller.js (hunk 1 above)
Comment (robwalch) [reply src/js/controller/controller.js:437]:
```
            const adState = _getAdState();
            if (adState && adState !== STATE_PAUSED) {
               _updatePauseReason(meta);
               if (adState === STATE_BUFFERING) {
                    // TODO: defer call to pauseAd
                    ...
               } else {
                    _api.pauseAd(true, meta);
               }
                return;
            }
```

File: src/js/controller/controller.js [hunk 2]
```diff
@@ -431,16 +431,34 @@ Object.assign(Controller.prototype, {
...
+                    _this._instreamAdapter.noResume = !viewable;
```
Comment (robwalch) [inline src/js/controller/controller.js:438]: `!viewable` is always `true` here.

**Comment by jwrobot-player** [comment]
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1842/) MULTI Build for commit 858e13cb05a5ed03906edfbc094e300e19070053 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9248/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9248/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6209/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6209/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28186/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28186/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23662/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23662/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/5855/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28186/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28186/archive/test/view/) Views

**Review by robwalch — APPROVED** [review]

**Pull request merged**